include/kdbush.hpp

```diff
@@ -1,64 +1,89 @@
+// https://github.com/mourner/kdbush.hpp
+
 #pragma once
 
 #include <algorithm>
 #include <cmath>
 #include <vector>
 
+template<class TFloat = double, class TIndex = std::size_t>
 class KDBush {
+    
+public:
 
-    using size_t = std::size_t;
+    static const uint8_t defaultNodeSize = 64;
+    
+    KDBush(const std::vector<TFloat>& coords, const uint8_t nodeSize_ = defaultNodeSize)
+        : KDBush(std::begin(coords), std::end(coords), nodeSize_) {}
 
-public:
-    KDBush(const std::vector<double> &coords_, const uint8_t nodeSize_ = 64)
-        : coords(coords_), nodeSize(nodeSize_) {
+    template<class TFloatIter>
+    KDBush(TFloatIter coordsBegin, TFloatIter coordsEnd, const uint8_t nodeSize_ = defaultNodeSize)
+        : coords(std::distance(coordsBegin, coordsEnd)), nodeSize(nodeSize_) {
+
+        std::copy(coordsBegin, coordsEnd, coords.data());
 
         const auto ids_size = coords.size() / 2;
         ids.reserve(ids_size);
-        for (size_t i = 0; i < ids_size; i++) ids.push_back(i);
+        for (size_t i = 0; i < ids_size; i++) ids.push_back(TIndex(i));
```

Comment:
We can add static assertions in there that require `TIndex` to be at least X bit or so